### PR TITLE
Draft: Made API's to connect to Google Calendar and generate tokens.

### PR DESCRIPTION
**Pull request by yesyash**
### Note: This is just a draft pr, the goal is to show a working concept of the user being able to authorize their google calendar.

### CODE REVIEW

File: routes/calendar.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const googleConnectHandler = (_: Request, res: Response): void => {
+  const url = oauth2Client.generateAuthUrl({
+    access_type: "offline",
+    scope: scopes,
+  });
+
+  res.redirect(url);
+};
+
+const googleCallbackHandler = async (
```
Comment (ankurnarkhede) [inline routes/calendar.ts:24]: We can add this in controller.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "build": "npx tsc",
     "prestart": "npm run build",
     "start": "node dist/server.js",
-    "dev": "NODE_ENV=development tsc-watch --onSuccess \"node ./dist/server.js\"",
+    "dev": "tsc-watch --onSuccess \"node ./dist/server.js\"",
```
Comment (ankurnarkhede) [inline package.json:13]: Why have we removed this?

File: routes/calendar.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+): Promise<void> => {
+  const query = req.query as { code: string; scope: string };
+
+  /**
+   * Generate AccessTokens from Secret code
+   * Note: We only get Refresh token when the user gives permission for the first time
+   * */
+  const { tokens } = await oauth2Client.getToken(query.code);
+
+  res.json({
```
Comment (ankurnarkhede) [inline routes/calendar.ts:36]: Add a consistent return before res.json()

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (yesyash) [reply package.json:13]: This was causing error on my windows machine and hence is temporarily removed just for this draft pr.

Comment (ankurnarkhede) [reply package.json:13]: We will need to use the cross-env lib for this. Can you add it?

**Pull request closed without merging**